RIOT-OS/RIOT
# boards/lpc1768: move vendor conf code to CPU level

### Contribution description

This PR moves lpc1768 CPU configuration code of mbed_lpc1768 and seeduino_arch-pro boards from board level to cpu level. The `vendor_conf.h` and `system.c` and strictly the same for both boards, so this PR removes duplicated code.

Since `vendor_conf.h` is moved to the `cpu/lpc1768/include/vendor` directory, the broken header guard of that file doesn't trigger a failure.

CPU configurability (for clock, etc) could be improved (using `#ifndef/#endif` and eventually Kconfig definitions) but I leave this as follow-up PRs.




### Testing procedure

- Green CI
- Firmware sizes for both boards should remain unchanged. Here are examples with `examples/hello-world`:

<details><summary>mbed_lpc1768</summary>

- master

```
$ RIOT_CI_BUILD=1 make BOARD=mbed_lpc1768 -C examples/hello-world clean all --no-print-directory 
Building application "hello-world" for "mbed_lpc1768" with MCU "lpc1768".

   text	   data	    bss	    dec	    hex	filename
   8488	    112	   2304	  10904	   2a98	/work/riot/RIOT/examples/hello-world/bin/mbed_lpc1768/hello-world.elf
```

- this PR

```
$ RIOT_CI_BUILD=1 make BOARD=mbed_lpc1768 -C examples/hello-world --no-print-directory 
Building application "hello-world" for "mbed_lpc1768" with MCU "lpc1768".

   text	   data	    bss	    dec	    hex	filename
   8488	    112	   2304	  10904	   2a98	/work/riot/RIOT/examples/hello-world/bin/mbed_lpc1768/hello-world.elf
```

</details>

<details><summary>seeeduino_arch-pro</summary>

- master

```
$ RIOT_CI_BUILD=1 make BOARD=seeeduino_arch-pro -C examples/hello-world clean all --no-print-directory 
Building application "hello-world" for "seeeduino_arch-pro" with MCU "lpc1768".

   text	   data	    bss	    dec	    hex	filename
   8500	    112	   2304	  10916	   2aa4	/work/riot/RIOT/examples/hello-world/bin/seeeduino_arch-pro/hello-world.elf
```

- this PR

```
$ RIOT_CI_BUILD=1 make BOARD=seeeduino_arch-pro -C examples/hello-world --no-print-directory 
Building application "hello-world" for "seeeduino_arch-pro" with MCU "lpc1768".

   text	   data	    bss	    dec	    hex	filename
   8500	    112	   2304	  10916	   2aa4	/work/riot/RIOT/examples/hello-world/bin/seeeduino_arch-pro/hello-world.elf
```

</details>

- broken headerguard is not reported anymore by the check script:

<details><summary>master</summary>

```
$ ./dist/tools/headerguards/check.sh 
boards/mbed_lpc1768/include/vendor_conf.h: no / broken header guard
boards/seeeduino_arch-pro/include/vendor_conf.h: no / broken header guard
sys/posix/include/fcntl.h: no / broken header guard
```

</details>

<details><summary>this PR</summary>

```
$ ./dist/tools/headerguards/check.sh 
sys/posix/include/fcntl.h: no / broken header guard
```

</details>




### Issues/PRs references

Should be merged before #15740
